integration/lsp/test_utils.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {fork} from 'child_process';
+import * as fs from 'fs';
+import {resolve} from 'path';
+import {createMessageConnection, IPCMessageReader, IPCMessageWriter, MessageConnection} from 'vscode-jsonrpc';
+import * as lsp from 'vscode-languageserver-protocol';
+
+const SERVER_PATH = resolve(__dirname, '../../../dist/server/index.js');
+const PACKAGE_ROOT = resolve(__dirname, '../../..');
+const PROJECT_PATH = resolve(__dirname, '../../project');
+export const APP_COMPONENT = `${PROJECT_PATH}/app/app.component.ts`;
+export const FOO_TEMPLATE = `${PROJECT_PATH}/app/foo.component.html`;
+
+export interface ServerOptions {
+  ivy: boolean;
+}
+
+export function createConnection(serverOptions: ServerOptions): MessageConnection {
+  const argv: string[] = [
+    '--node-ipc',
+    '--tsProbeLocations',
+    PACKAGE_ROOT,
+    '--ngProbeLocations',
+    SERVER_PATH,
+  ];
+  if (serverOptions.ivy) {
+    argv.push('--experimental-ivy');
+  }
+  const server = fork(SERVER_PATH, argv, {
+    cwd: PROJECT_PATH,
+    env: {
+      TSC_NONPOLLING_WATCHER: 'true',
+    },
+  });
+
+  return createMessageConnection(
+      new IPCMessageReader(server),
+      new IPCMessageWriter(server),
+  );
+}
+
+export function initializeServer(client: MessageConnection): Promise<lsp.InitializeResult> {
+  return client.sendRequest(lsp.InitializeRequest.type, {
+    /**
+     * The process Id of the parent process that started
+     * the server. Is null if the process has not been started by another
+     * process. If the parent process is not alive then the server should
+     * exit (see exit notification) its process.
+     */
```

Comment:
Is this comment copied from the specification or somewhere else? I think we can simplify it to just something like
```suggestion
    /**
     * The process id of the parent process that started
     * the server. Is always the current process.
     */
```
for our use.